File: internal/implant/extensions/testdata/script/hello.lua
#!/usr/bin/env lua

-- Get arguments from environment
local args_json = os.getenv("EXTENSION_ARGS") or "{}"

-- Simple JSON parsing (just extract name)
local name = args_json:match('"name"%s*:%s*"([^"]+)"') or "World"

-- Generate output
local output = string.format([[{
    "success": true,
    "output": "Hello, %s! This is Lua extension.",
    "exit_code": 0,
    "data": {
        "lua_version": "%s",
        "args_received": %s
    }
}]], name, _VERSION, args_json)

-- Print result
print(output)
os.exit(0)

File: internal/implant/extensions/testdata/script/hello.js
#!/usr/bin/env node

// Get arguments from environment
const args = JSON.parse(process.env.EXTENSION_ARGS || '{}');

// Get name argument
const name = args.name || 'World';

// Generate output
const output = {
    success: true,
    output: `Hello, ${name}! This is JavaScript extension.`,
    exit_code: 0,
    data: {
        node_version: process.version,
        args_received: args
    }
};

// Print result as JSON
console.log(JSON.stringify(output));
process.exit(0);